docs/tests.md

```diff
@@ -0,0 +1,415 @@
+# Tests
+
+## Testing Suite Architecture Overview
+
+This section provides an overview of the testing architecture for our Python
+library, which employs two types of tests: **regression tests** and
+**doctests**. Both types are powered by pytest, with doctests **(WIP)** being
+specifically based on the Python's doctest module.
+
+## Regression Tests: Understanding the Process
+
```

Comment:
I would define this tests differently to highlight that the estimates compute by the estimators library somehow match those made by state of the art of estimators. 

I would suggest to include something like this:
"The goal of these tests is to ensure that the complexity estimates made by the `CryptographicEstimators` library match the ones produced by state-of-the-art estimators implemented by researchers of the cryptographic community."